documentation/en/chain-indexer-overview-for-operators.md

```diff
@@ -0,0 +1,415 @@
+# supporting functionality that will now be running.)
+  EnableIndexer = true 
+```
+
+You can learn more about these configuration options and other configuration options available for the `ChainIndexer` [here](https://github.com/filecoin-project/lotus/blob/master/documentation/en/default-lotus-config.toml).
+
+
+### Garbage Collection
+
+The `ChainIndexer` includes a garbage collection (GC) mechanism to manage the amount of historical data retained.  See the [ChainIndexer size requirements](#backfill-disk-space-requirements).
+
+By default, GC is disabled to preserve all indexed data.
+
+To configure GC, use the `GCRetentionEpochs` parameter in the `ChainIndexer` section of your config.
+
+The ChainIndexer [periodically runs](https://github.com/filecoin-project/lotus/blob/master/chain/index/gc.go#L15) GC if `GCRetentionEpochs` is > 0 and removes indexed data for epochs older than `(current_head_height - GCRetentionEpochs)`.
+
+```toml
+[ChainIndexer]
+  GCRetentionEpochs = X  # Replace X with your desired value
+```
+
+- Setting `GCRetentionEpochs` to 0 (**default**) disables GC.
+- Any positive value enables GC and determines the number of epochs of historical data to retain.
+
+#### Recommendations
+
+1. **Archival Nodes**: **Keep GC disabled** (`GCRetentionEpochs` = 0) to retain all indexed data.
+
+2. **Non-Archival Nodes**:  Set `GCRetentionEpochs` to match the amount of chain state your node retains 
+
+**Example:** if your node is configured to retain Filecoin chain state with a Splitstore Hotstore that approximates 2 days of epochs, set `GCRetentionEpochs` to at least `retentionDays * epochsPerDay = 2 * 2880 = 5760`).
+
+**Warning:** Setting this value below the chain state retention period may degrade RPC performance and reliability because the ChainIndexer will lack data for epochs still present in the chain state.
+
+**Note:** `Chainstore.Splitstore` is configured in terms of bytes (not epochs) and `ChainIndexer.GCRetentionEpochs` is in terms of epochs (not bytes).  For the purposes of this discussion, we're assuming operators have determined `Chainstore.Splitstore.HotStoreMaxSpaceTarget` and `Chainstore.Splitstore.HotStoreMaxSpaceThreshold` values that approximate a certain number days of storage in the Splitstore Hotstore.  The guidance here is to make sure this approximation exceeds `ChainIndexer.GCRetentionEpochs`.
+
+### Removed Options
+
+**Note: The following config options no longer exist in Lotus and have been removed in favor of the ChainIndexer config options explained above. They can be removed when upgrading to Lotus v1.31.0.**
+
+```toml
+[Fevm]
+EthTxHashMappingLifetimeDays
+
+[Events]
+DisableRealTimeFilterAPI
+DisableHistoricFilterAPI
+DatabasePath
+
+[Index]
+EnableMsgIndex
+```
+
+The previously deprecated `Fevm.Events` options are now also all removed:
+* `Fevm.Events.DatabasePath`
+* `Fevm.Events.DisableRealTimeFilterAPI`
+* `Fevm.Events.DisableHistoricFilterAPI`
+* `Fevm.Events.FilterTTL`
```

Comment:
The `FilterTTL`, `MaxFilters`, `MaxFilterResults` AND `MaxFilterHeightRange` options have NOT been removed and are still applicable as the `ChainIndexer` does not make any changes to the ETH Event Filter APIs.

Let's fix this.

Reply:
They've been removed from `Fevm.Events`, that's the deprecated (`moved`) block that got deleted.

Reply:
Ah okay, they do still exist in `EventsConfig`. 

So might be worth calling out that the **_only_** way to configure these now might be `EventsConfig. FilterTTL`, `EventsConfig.MaxFilters` etc .

Reply:
I don't know the differences enough to write this.  Can you make a proposal or change @aarshkshah1992 ?  (or have quick sync with me so I can come up to speed)

Reply:
I think I fixed this in https://github.com/filecoin-project/lotus/pull/12600/commits/8463a03345d8c28410825d0cc115224976522d31 .  That said, I'm confused why we support it in one area and not in the other.

Or is this a case that these are good config options but they never should have been exposed through `Fevm.Events` and we're now cleaning that up?  

I don't know enough here, so please feel free to take over the wording here.

Reply:
@BigLep they were all originally only on `Fevm.Events` but I moved them to a new top-level `Events` when we introduced builtin actor events. At that time, instead of hard-removing them, I introduced a soft deprecation method, where I deprecated them in documentation so they don't show up and they have a "moved" tag associated with them so when someone starts their node using one of these settings, it sets the right one automatically for them and complains in their logs that it's deprecated and they should fix it.

So with this release, we're deciding to hard-deprecate and just remove them, no more complaints, no more automatic setting of the right value. I'll add a suggestion to clear it up a little more.